src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -29,23 +29,23 @@ class DefaultChoiceListFactory implements ChoiceListFactoryInterface
     /**
      * {@inheritdoc}
      */
-    public function createListFromChoices($choices, $value = null)
+    public function createListFromChoices(iterable $choices, callable $value = null)
     {
         return new ArrayChoiceList($choices, $value);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function createListFromLoader(ChoiceLoaderInterface $loader, $value = null)
+    public function createListFromLoader(ChoiceLoaderInterface $loader, callable $value = null)
     {
         return new LazyChoiceList($loader, $value);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function createView(ChoiceListInterface $list, $preferredChoices = null, $label = null, $index = null, $groupBy = null, $attr = null)
+    public function createView(ChoiceListInterface $list, $preferredChoices = null, callable $label = null, callable $index = null, callable $groupBy = null, $attr = null)
```

Comment:
this change breaks TwigBundle, which passes a `false` label:
> TypeError: Argument 3 passed to Symfony\Component\Form\ChoiceList\Factory\DefaultChoiceListFactory::createView() must be callable or null, boolean given, called in /home/travis/build/symfony/symfony/src/Symfony/Component/Form/ChoiceList/Factory/PropertyAccessDecorator.php on line 193

Reply:
See #33074